### ISSUE TITLE
Add option to dump CompileExpr in a standard, parseable format

### ISSUE THREAD

**Issue by evertedsphere** [edwinb/Idris2-boot#278]
This seems fairly simple, and I intend to add support for this myself. The easiest way, if not the "correct" one, is to add a dummy backend that pretty-prints the CompileExpr it receives to a file.

**Comment by christian-public** [comment]
I think `--dumpcases` is what you are looking for. It was added in commit 59503712.

If you have a `Main.idr` with a `main : IO ()` function, you can use `--dumpcases` like this:
```
idris2 --dumpcases foo -o bar Main.idr
```

- `foo` is the name of the file where the CompileExpr will be dumped (located in the current directory)
- `bar` is the name of the output executable (located in `build/exec`)

**Comment by evertedsphere** [comment]
I don't think so; I guess I should've mentioned that my use-case requires the output to be easily parseable in a way that the output of `--dumpcases` is not, so I can use it as an input for further processing.

I discussed this with @edwinb yesterday, and it seems we may want to standardise a stable s-expression-style text format for this.
